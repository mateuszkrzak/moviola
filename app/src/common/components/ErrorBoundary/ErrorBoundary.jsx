import React from 'react';
import PropTypes from 'prop-types';

import styles from './ErrorBoundary.scss';

class ErrorBoundary extends React.Component {
  static defaultProps = {
    children: <div>No children passed to ErrorBoundary</div>,
  };

  static propTypes = {
    children: PropTypes.node,
  };

  state = { error: '', info: '' };

  componentDidCatch(error, info) {
    this.setState({ error, info });
  }

  render() {
    if (this.state.error !== '') {
      return (
        <div className="error">
          <div className="container">
            <span>
              There was error:
              <pre>{this.state.error.message}</pre>
            </span>
            <span>and it occured {this.state.info.componentStack}.</span>
          </div>
        </div>
      );
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
