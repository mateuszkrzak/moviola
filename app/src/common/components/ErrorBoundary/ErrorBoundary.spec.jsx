import { shallow } from 'enzyme';
import React from 'react';
import ErrorBoundary from './ErrorBoundary';

class BuggedComponent extends React.Component {
  makeBug = () => {
    throw new Error('Test error message');
  };

  render() {
    this.makeBug();

    return <span>This should not appear anywhere</span>;
  }
}

test('ErrorBoundary generates a error message when an error is caught', () => {
  const BuggedComponent = () => {
    throw new Error('Test error message');
  };
  const wrapper = shallow(<ErrorBoundary>
    <BuggedComponent />
                          </ErrorBoundary>);

  expect(wrapper).toMatchSnapshot();
});
