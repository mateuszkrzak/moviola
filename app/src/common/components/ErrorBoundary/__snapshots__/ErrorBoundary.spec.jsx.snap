// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary generates a error message when an error is caught 1`] = `<BuggedComponent />`;
